presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -170,38 +189,42 @@ public boolean equals(Object o)
         }
         PlanNodeStatsEstimate that = (PlanNodeStatsEstimate) o;
         return Double.compare(outputRowCount, that.outputRowCount) == 0 &&
+                Double.compare(totalSize, that.totalSize) == 0 &&
                 Objects.equals(variableStatistics, that.variableStatistics);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(outputRowCount, variableStatistics);
+        return Objects.hash(outputRowCount, totalSize, variableStatistics);
     }
 
     public static Builder builder()
     {
         return new Builder();
     }
 
+    // Do not inherit totalSize statistics by default. Only explicitly set
     public static Builder buildFrom(PlanNodeStatsEstimate other)
     {
-        return new Builder(other.getOutputRowCount(), other.variableStatistics);
+        return new Builder(other.getOutputRowCount(), NaN, other.variableStatistics);
```

Comment:
I'm a bit against propagating the `totalSize` by default for the current scope of this PR, because people have to explicitly set it to `NaN` or otherwise a not `NaN` but possibly wrong `totalSize` value is carried. Given that we are only using this field for "leaf" simple join plans for now, it is safer to set it `NaN` by default to so that if people "accidentally" fetch the totalSize at other places they can tell that it is not usable via `isNaN()`.

Ideally, when we have implemented how to handle the totalSize field for all types of operator rules, we could set propagate as default to simply the relevant operations in those rules that won't change this field.

What do you think?

Reply:
that makes senes to me

Reply:
Could you please include this explanation in the comment?